bankroll/converter/modelconverter.py

```diff
@@ -0,0 +1,74 @@
+from typing import Any, Callable, cast, List, Optional, Sequence, Tuple, Union
+
+try:
+    import pandas as pd
+except ImportError:
+    pass
+
+from bankroll.model import Position, Trade
+
+ConvertibleModel = Union[Position, Trade]
+ColumnFunctions = List[Tuple[str, Callable[[Any], Any]]]
+
+
+def dataframeForModelObjects(
+    items: Sequence[ConvertibleModel]
+) -> Optional[pd.DataFrame]:
+    df = None
+
+    if module_exists("pandas"):
+        rows: List[List[Any]] = []
+        firstItem: Optional[ConvertibleModel] = None
+        for i in items:
+            if firstItem == None:
+                firstItem = i
+            else:
+                assert type(firstItem) == type(i)
+
+            rows.append([t[1](i) for t in dataframeColumnFunctions(i)])
+
+        if firstItem is not None:
+            actualFirstItem: ConvertibleModel = firstItem
+            columns = [t[0] for t in dataframeColumnFunctions(actualFirstItem)]
+            df = pd.DataFrame(rows, columns=columns)
+
+    return df
+
+
+def dataframeColumnFunctions(model: ConvertibleModel) -> ColumnFunctions:
+    if isinstance(model, Position):
+        posInstrument: Callable[[Position], Any] = lambda p: str(p.instrument)
```

Comment:
I'm guessing you wrote all these temporary variables to make the types unify.

I _think_ you should be able to achieve the same thing by just saving the dictionary into a local:

```py
cf: ColumnFunctions = {
  'Instrument': lambda p: str(p.instrument),
  'Quantity': lambda p: p.quantity.normalize()
  ...
}

return cf
```